### DISCUSSION TITLE
terragrunt destroy -target with before_hook present

### DISCUSSION THREAD

**Discussion by js-max** [gruntwork-io/knowledge-base#594] [How do I]
Hi all,

I already went through 'Discussions' list/search box and I was not able to clarify my doubt.

```bash
─ stack
 ├── 01_folder
 ├── 02_folder
 └── 03_folder
```

Using the basic structure I will configure a dependency between folder **02** and **03**.
Also I need to cleanup some resources using `terragrunt state rm X` when tearing down everything.

```hcl
# terragrunt.hcl in 03_folder
...
dependency "02" {
  config_path = "../02_folder"
}
...
terraform {
  before_hook "remove_apps" {
    commands = ["destroy"]
    execute  = ["sh", "-c", "for i in $(terragrunt state list | grep -E 'kubernetes_secret|helm_release'); do terragrunt state rm $i; done"]
  }
}
...
```

So far so good.. It's working fine when I run the basic `terragrunt run-all destroy` CLI.

The issue arrives when within **02_folder** I want to perform `terragrunt destroy -target resource.X` for that specific folder and `terragrunt` starts to execute `before_hook` present within **03_folder** due to `dependency`.

Of course, that will damage/break the stack as we just wanted to remove a specific `-target` within **02_folder**.

Since I'm using Terragrunt I was looking to work with `dependency` instead of Terraform `data`.

Any suggestion to overcome this?

Thanks!

---

<ins datetime="2022-11-18T16:46:15Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109619">Tracked in ticket #109619</a></p>
</ins>

**Comment by bethadele** [comment]
I'm checking with one of our Terragrunt experts to get some insight into the issue.

**Comment by denis256** [comment]
Hi,
AFAIK conditional execution of hooks in Terragrunt is not supported but can be implemented workarounds by defining environment variables to allow execution of hook or not

**Reply by js-max** [reply]
Hi @denis256 , is it possible to have an example from your side for that workaround?
